geode-core/src/main/java/org/apache/geode/cache/query/IndexType.java

```diff
@@ -22,7 +22,7 @@
  *
  * @since GemFire 4.0
  */
-public class IndexType {
+public enum IndexType {
```

Comment:
Is it backwards compatible to change a class to an enum? This class isn't sent over the wire, but if someone compiled code against an older version of geode, would that code break if it ran against a newer version?

Reply:
Yes, it is binary compatible. Basically we are Inserting new class or interface types in the type hierarchy, and from here https://docs.oracle.com/javase/specs/jls/se7/html/jls-13.html it's listed as one of the binary compatible operations.

Reply:
I believe that change is not backwards compatible in terms of serialization (http://docs.oracle.com/javase/6/docs/platform/serialization/spec/version.html#6678).  

On the other hand, since the only public API exported by the class was the set of `public static final` constants and the same syntax will now refer to the enum values, I think this is backwards compatible in terms of compilation.

Reply:
Interesting. This class wasn't Serializable, so we are ok there.

I was concerned that a class compiled with the old version would throw a NoSuchField exception or some such if something about the fields changed by making this an enum. But a ran a small test and it looks like it does work ok. Looks good!

Reply:
This IndexType was not Serializable anyway and is not sent across the wire. This is compatible in the sense that code compiled with the old version of IndexType running with the new version.